packages/tailwindcss/src/index.ts

```diff
@@ -61,15 +61,49 @@ async function parseCss(
 
     // Collect paths from `@plugin` at-rules
     if (node.selector === '@plugin' || node.selector.startsWith('@plugin ')) {
-      if (node.nodes.length > 0) {
-        throw new Error('`@plugin` cannot have a body.')
-      }
-
       if (parent !== null) {
         throw new Error('`@plugin` cannot be nested.')
       }
 
-      pluginPaths.push(node.selector.slice(9, -1))
+      let options: CssPluginOptions = {}
+
+      for (let decl of node.nodes ?? []) {
+        if (decl.kind !== 'declaration') {
+          throw new Error('`@plugin` can only contain declarations.')
+        }
+
+        if (decl.value === undefined) continue
+
+        // Parse the declaration value as a primitive type
+        // These are the same primitive values supported by JSON
+        let value: CssPluginOptions[keyof CssPluginOptions] = decl.value
+
+        if (value === 'null') {
+          value = null
+        } else if (value === 'true') {
+          value = true
+        } else if (value === 'false') {
+          value = false
+        } else if (!Number.isNaN(Number(value))) {
+          value = Number(value)
+        } else if (value[0] === '"' && value[value.length - 1] === '"') {
+          value = value.slice(1, -1)
+        } else if (value[0] === "'" && value[value.length - 1] === "'") {
+          value = value.slice(1, -1)
+        } else if (value[0] === '[' && value[value.length - 1] === ']') {
+          throw new Error('Arrays are not supported in `@plugin` options.')
+        } else if (value[0] === '{' && value[value.length - 1] === '}') {
+          throw new Error('Objects are not supported in `@plugin` options.')
+        }
```

Comment:
Maybe those two errors could also say the property that this error was found and maybe give some options? (just brainstorming here though).

> Unexpected `@plugin` option. Value for declaration `foo: [...blabla]` is not supported. It looks like you want to pass an array to plugin options. This is not supported from CSS. Instead, create a custom JS plugin file: 
> 
> ``` 
> // ./my-plugin.js
> export default ...
> ```

(Ok the thing at the end should probably be a docs link)